index.js

```diff
@@ -516,13 +516,23 @@ Line2D.prototype.update = function (options) {
 					pos[ptr++] = y
 				}
 
-				let triangles = triangulate(pos, state.hole || [])
-
-				for (let i = 0, l = triangles.length; i < l; i++) {
-					if (ids[triangles[i]] != null) triangles[i] = ids[triangles[i]]
+				if(!(state.count-1 in ids))
+					ids[state.count] = state.count-1   // make sure there is at least polygon
+					
+				let splits = Object.keys(ids).map(e => e|0).sort(function(a, b){return a - b})
```

Comment:
What's the purpose of `map(e => e|0)`?
Why sort needs that default comparator?

Reply:
Yes, the lambda is for casting, will replace it with map(Number)
The comparator for sort is to sort by number value rather than string order.

Sorry I am not very fluent in javascript.